.golangci.yml

```diff
@@ -0,0 +1,17 @@
+issues:
+        max-same-issues: 0
+        exclude-use-default: false
+linters:
+        enable:
+        - whitespace
+        - gosec
+        - gci
+        - misspell
+        - gomnd
```

Comment:
wouldn't this require changing *tons* of lines? I'm not sure if it's worth the effort

Reply:
Please, feel free to propose the golangci.yml file that you think we should have.

Regarding using a golang-ci linter I do see a lot of possible future value here. Most of the reviews we do on a PR would be automatically find by the linter so that would save a lot of time doing always the same kind of review (empty line not necessary, imports not cgi-ed, error not check, variable not needed, etc.). Currently we only have staticcheck (which is nice) but not enough.

Said that I am very open to suggestion on which would be the best tool or the best configuration for golang-ci.

Reply:
I made some suggestions for linters in the other comments. I agree that using more linters is a good thing. I don't like using golangci-lint because its quality is questionable as a tool (see my link from earlier), and at the end of the day I only end up running two or three linters on my projects.

Reply:
Then I'll apply your proposals and lets try how it goes... we can always change.

Reply:
I'm confused. This PR is open, but golangci-lint is on master without these changes applied.

Reply:
Yes. This PR will never be merged, its just a "reminder". The idea is that we'll perform commits time to time in order to apply the fixes (not a single MR but several).

I decided to merge golangci.yml so all of us making the suggestions can use the same config options.

Reply:
I have updated it with your comments.